rest-service/manager_rest/resources_v2.py

```diff
@@ -92,6 +93,204 @@ def _create_filter_params_list_description(parameters, list_type):
              'paramType': 'query'} for filter_val in parameters]
 
 
+def _get_snapshot_path(snapshot_id):
+    return os.path.join(
+        config.instance().file_server_root,
+        config.instance().file_server_snapshots_folder,
+        snapshot_id
+    )
+
+
+class UploadedSnapshotsManager(files.UploadedDataManager):
+
+    def _get_kind(self):
+        return 'snapshot'
+
+    def _get_data_url_key(self):
+        return 'snapshot_archive_url'
+
+    def _get_target_dir_path(self):
+        return config.instance().file_server_snapshots_folder
+
+    def _get_archive_type(self, archive_path):
+        return 'zip'
+
+    def _prepare_and_process_doc(self, data_id, file_server_root,
+                                 archive_target_path):
+        return get_blueprints_manager().create_snapshot_model(
+            data_id,
+            status=models.Snapshot.UPLOADED
+        ), None
+
+
+class Snapshots(SecuredResource):
+
+    @swagger.operation(
+        responseClass='List[{0}]'.format(responses_v2.Snapshot.__name__),
+        nickname='list',
+        notes='Returns a list of existing snapshots.'
+    )
+    @exceptions_handled
+    @marshal_with(responses_v2.Snapshot)
+    def get(self, _include=None):
+        return get_blueprints_manager().snapshots_list(_include)
+
+
+class SnapshotsId(SecuredResource):
+
+    @swagger.operation(
+        responseClass=responses_v2.Snapshot,
+        nickname='getById',
+        notes='Returns a snapshot by its id.'
+    )
+    @exceptions_handled
+    @marshal_with(responses_v2.Snapshot)
+    def get(self, snapshot_id, _include=None, **kwargs):
+        return get_blueprints_manager().get_snapshot(snapshot_id,
+                                                     include=_include)
+
+    @swagger.operation(
+        responseClass=responses_v2.Snapshot,
+        nickname='createSnapshot',
+        notes='Create a new snapshot of the manager.',
+        consumes=[
+            "application/json"
+        ]
+    )
+    @exceptions_handled
+    @marshal_with(responses_v2.Execution)
+    def put(self, snapshot_id):
+        verify_json_content_type()
+        request_json = request.json
+        verify_parameter_in_request_body('include_metrics', request_json)
```

Comment:
Verification can include the parameter type (`param_type` argument).
Perhaps we should have default values for these parameters and make them optional?